ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -998,27 +998,37 @@ getDocsBatch
   :: HscEnv
   -> Module  -- ^ a moudle where the names are in scope
   -> [Name]
-  -> IO [Either String (Maybe HsDocString, Map.Map Int HsDocString)]
+  -> IO [(Name, Either String (Maybe HsDocString, Map.Map Int HsDocString))]
 getDocsBatch hsc_env _mod _names = do
```

Comment:
tl;dr Providing a proper pairing of `Names` to their `Docs` here.

---

In the:
```haskell
-> [Name]
-> IO [Either String (Maybe HsDocString, Map.Map Int HsDocString)]
```
We see that usage of the function also implicitly required for dev/code to do outside match of the indexes between `[names]` & their `[docs]`.

But soon the code would be also processing the `Map.Map Int HsDocString` (`ArgDocs`) part, so creating a maps for docs would be much simpler & effective than having:
Instead of `[Name] -> [(Maybe HsDocString, Map.Map Int HsDocString)]`, docs would be retrieved as `Map Name (Maybe HsDocString, IntMap HsDocString)`, where `Maybe HsDocString` is a function doc block & `IntMap HsDocString` is args docs (the eventual type there is to be seen in the future, when we finish arg docs provisioning, so currently keeping what was used in the code before).